source/ScrollSync/ScrollSync.js

```diff
@@ -1,20 +1,50 @@
-import PropTypes from 'prop-types';
+// @flow
 import * as React from 'react';
 
+type Scrollable = {
+  +scrollToPosition: ({scrollTop: number, scrollLeft: number}) => void,
+};
+
+type ScrollableElement = ?React.ElementRef<any> & Scrollable;
+
+type OnScrollParams = {
+  clientHeight: number,
+  clientWidth: number,
+  scrollHeight: number,
+  scrollLeft: number,
+  scrollTop: number,
+  scrollWidth: number,
+};
+
+type Props = {
+  /**
+   * Function responsible for rendering 2 or more virtualized components.
+   */
+  children: (
+    params: OnScrollParams & {onScroll: OnScrollParams => void} & {
+      registerChild: (element: ScrollableElement | null) => void,
+    },
+  ) => React.Node,
+};
+
+type State = {
+  clientHeight: number,
+  clientWidth: number,
+  scrollHeight: number,
+  scrollLeft: number,
+  scrollTop: number,
+  scrollWidth: number,
+};
+
 /**
  * HOC that simplifies the process of synchronizing scrolling between two or more virtualized components.
  */
-export default class ScrollSync extends React.PureComponent {
-  static propTypes = {
-    /**
-     * Function responsible for rendering 2 or more virtualized components.
-     * This function should implement the following signature:
-     * ({ onScroll, scrollLeft, scrollTop }) => PropTypes.element
-     */
-    children: PropTypes.func.isRequired,
-  };
+export default class ScrollSync extends React.PureComponent<Props, State> {
+  _elements: (ScrollableElement | null)[] = [];
+  _onScroll: (params: OnScrollParams) => void;
+  _registerChild: (element: ScrollableElement | null) => void;
 
-  constructor(props, context) {
+  constructor(props: Props, context: any) {
```

Comment:
nit: can we get the contextTypes

Reply:
Yes, I would like to, but I'm struggling with this one a bit since I don't have much experience with flow and legacy context 😅 I'm not sure what should I exactly do, this context here is just passed to the parent.